### PR TITLE
Add types for sumoselect

### PR DESCRIPTION
**Pull request by nagamejun**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/sumoselect/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    csvSepChar: string;
+    okCancelInMulti: boolean;
+    isClickAwayOk: boolean;
+    triggerChangeCombined: boolean;
+    selectAll: boolean;
+    search: boolean;
+    searchText: string;
+    noMatch: string;
+    prefix: string;
+    locale: any[];
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:28]: this should be an array of strings

https://github.com/HemantNegi/jquery.sumoselect/blob/master/jquery.sumoselect.js#L49

File: types/sumoselect/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Type definitions for sumoselect 3.0
+// Project: https://github.com/HemantNegi/jquery.sumoselect#readme
+// Definitions by: nagamejun <https://github.com/nagamejun>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="jquery" />
+
+type SumoSelectEvents = 'sumo:opening' | 'sumo:opened' | 'sumo:closing' | 'sumo:closed';
+
+type Option = Partial<{
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:10]: missing `searchFn`

https://github.com/HemantNegi/jquery.sumoselect/blob/master/jquery.sumoselect.js#L44

File: types/sumoselect/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    isClickAwayOk: boolean;
+    triggerChangeCombined: boolean;
+    selectAll: boolean;
+    search: boolean;
+    searchText: string;
+    noMatch: string;
+    prefix: string;
+    locale: any[];
+    up: string;
+    showTitle: string;
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:30]: this should be boolean

https://github.com/HemantNegi/jquery.sumoselect/blob/master/jquery.sumoselect.js#L50

```suggestion
    showTitle: boolean;
```

File: types/sumoselect/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    okCancelInMulti: boolean;
+    isClickAwayOk: boolean;
+    triggerChangeCombined: boolean;
+    selectAll: boolean;
+    search: boolean;
+    searchText: string;
+    noMatch: string;
+    prefix: string;
+    locale: any[];
+    up: string;
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:29]: this should be boolean

https://github.com/HemantNegi/jquery.sumoselect/blob/master/jquery.sumoselect.js#L50

```suggestion
    up: boolean;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/sumoselect/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// Project: https://github.com/HemantNegi/jquery.sumoselect#readme
+// Definitions by: nagamejun <https://github.com/nagamejun>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="jquery" />
+
+type SumoSelectEvents = 'sumo:opening' | 'sumo:opened' | 'sumo:closing' | 'sumo:closed';
+
+type Option = Partial<{
+    placeholder: string;
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:11]: Please, could you provide a default value for each option? Example

```suggestion
/**
 * @default 'Select Here'
 */
    placeholder: string;
```

**Comment by nagamejun** [comment]
@martin-badin 
Thank you for the review.
I'm sorry for the my mistake...
Could you check the fix again?
e16b066...9c0b936

File: types/sumoselect/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+     */
+    search: boolean;
+    /**
+     * @default 'Search...'
+     */
+    searchText: string;
+    /**
+     * @default (haystack: string, needle: string) => boolean
+     */
+    searchFn: (...args: any[]) => boolean;
```
Comment (martin-badin) [inline types/sumoselect/index.d.ts:74]: this function has just two parameters

```suggestion
    searchFn(haystack: string, needle: string): boolean;
```

**Review by martin-badin — APPROVED** [review]
Thanks. Looks good to me :)

**Pull request merged**